app/renderer/components/Session/SavedSessions.js

```diff
@@ -19,9 +19,9 @@ export default class SavedSessions extends Component {
   onRow (record) {
     return {
       onClick: () => {
-        const {setCaps} = this.props;
+        const {setCapsWithServer} = this.props;
         let session = this.sessionFromUUID(record.key);
-        setCaps(session.caps, session.uuid);
+        setCapsWithServer(session.server, session.serverType, session.caps, session.uuid);
```

Comment:
we need to assume that someone who has an existing version of the app won't have any server or serverType defined, so let's make sure there's handling for that case somewhere (the default action should be to just do nothing in that case; keep the current server info as is; then when the user saves it again, the server info will get written)

Reply:
I was already mentioning this once that our save files must have versioned schemas. Not sure if it was ever implemented though

Reply:
Thanks for the help! I just added an if statement in SavedSessions.js file to check if capability set has a server and serverType defined before calling the setCapsAndServer function

Reply:
I assume we are going to have more and more `if`'s when new fields are added to saved files or existing ones are changed. A proper way of doing that stuff would be to have versioning and schemas/migrations.

Reply:
I'm not sure how much we want to future-proof this. A couple if statements are easier to read/understand than a schema migration system. I'd probably vote for keeping as-is for now and then revisiting if we find ourselves making lots of changes to this format (this is the first time in like 5 years we have done so).

Reply:
@mykola-mokhnach are you OK with this approach for now?